pkg/config/config.go

```diff
@@ -65,6 +65,62 @@ type MicroshiftConfig struct {
 	Ingress IngressConfig `json:"-"`
 }
 
+// Top level config
+type Config struct {
+	NodeName  string    `json:nodeName`
+	NodeIP    string    `json:nodeIP`
+	Network   Network   `json:network`
+	DNS       DNS       `json:dns`
+	Debugging Debugging `json:debugging`
+	ApiServer ApiServer `json:apiServer`
+}
+
+type ApiServer struct {
+	BindHost string `json:bindHost`
```

Comment:
The example for BindHost shows this as a URL, but the variable implies to me that I would set an IP or hostname.

Reply:
This variable name is proposed in the design doc that we discussed in the past. I'm happy to change it to something else more clear if we can agree on a new name. @benluddy

Reply:
Looking into it again, it seems that we are only using this option to set the cluster URL in the various kubeconfig files generated by MicroShift on intialization: https://github.com/openshift/microshift/blob/main/pkg/cmd/init.go#L364-L438

As such it feels a bit weird to have this option under apiserver when it is used globally to configure many components. Wouldn't it make more sense to keep this one as a top level option and rename it to something like `ServerURL` or `ClusterURL`?

cc @benluddy

Reply:
I synced with Ben as I forgot some context.

`BindHost` is the replacement for the original `URL` option which is used to set the server URL in the various kubeconfig files. However, we also noticed that this option was used to configure the port of the bind address of kube-apiserver: 
- https://github.com/openshift/microshift/blob/main/pkg/controllers/kube-apiserver.go#L108
- https://github.com/openshift/microshift/blob/main/pkg/config/config.go#L151-L170

The idea today is to change a bit the UX by having the kube-apiserver bind address explicitly configurable and generate the cluster URL for the kubeconfig files from it.

This wasn't the original way this PR was handling this option so we would have to update it. Also it might be worth renaming the option to `BindAddress` which is a bit clearer and follow what we have in OCP (https://github.com/openshift/api/blob/94ac871f4a586d03b0ae11344f3604016065a458/config/v1/types.go#L48).

Let me know if that sounds right to you @dhellmann.

Reply:
Discussed that briefly with @dhellmann offline, we will need an enhancement to change the existing behavior, so we will do that change in another PR.